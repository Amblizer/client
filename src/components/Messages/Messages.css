/* MESSENGER */

.Messages_MESSENGER {
  position: absolute;
  top: 32px;
  bottom: 48px;
  left: 0;
  right: 0;
  padding-top: 6px;
  overflow-y: scroll;
  border-right: 1px solid #cccccc;
  box-sizing: border-box;
}

.Messages-wrapper_MESSENGER {
  position: relative;
  height: 100%;
}

.Messages-list_MESSENGER {
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: 14px;
  font-family: sans-serif;
  width: 100%;
}

/* FLOAT */

.Messages_FLOAT {
  position: absolute;
  bottom: 48px;
  left: 0;
  right: 0;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
}

.Messages-wrapper_FLOAT {
  position: relative;
  height: 100%;
}

.Messages-list_FLOAT {
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: 14px;
  font-family: sans-serif;
  width: 100%;
}

/* SIDEPANEL */

.Messages_SIDEPANEL {
  position: absolute;
  top: 0;
  bottom: 48px;
  left: 0;
  right: 0;
  padding-top: 6px;
  /*border-right: 1px solid #cccccc;*/
  box-sizing: border-box;
}

.Messages-wrapper_SIDEPANEL {
  position: relative;
  height: 100%;
}

.Messages-list_SIDEPANEL {
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: 14px;
  font-family: sans-serif;
  width: 100%;
}