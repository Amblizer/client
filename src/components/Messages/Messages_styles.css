.Messages {
  position: absolute;
  top: 32px;
  bottom: 48px;
  left: 0;
  right: 0;
  padding-top: 6px;
  overflow-y: scroll;
  border-right: 1px solid #ccc;
  box-sizing: border-box;
}

.Messages-wrapper {
  position: relative;
  height: 100%;
}

.Messages-list {
  margin: 0;
  padding: 0;
  list-style: none;
  font-size: 14px;
  font-family: sans-serif;
  width: 100%;
}