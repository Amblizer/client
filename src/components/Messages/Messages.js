/* eslint react/jsx-filename-extension:0*/

import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import Message from '../Message/Message';
import './Messages_styles.css';


const MessageList = (props) => {
  const { socket, chatStyle } = props;

  const messages = props.messages.map(
    (message, index) => <Message key={index} author={message.author} content={message.content} />,
  );

  return (
    <div className={`Messages_${chatStyle}`}>
      <div className={`Messages-wrapper_${chatStyle}`}>
        <ul className={`Messages-list_${chatStyle}`}>
          {messages}
        </ul>
      </div>
    </div>
  );
        // <Notification />
        // <Status socket={socket} />
};

MessageList.propTypes = {
  socket: PropTypes.node,
  chatStyle: PropTypes.string,
  messages: PropTypes.array // eslint-disable-line
};

const messageListMapStateToProps = state => ({
  messages: state.chat.messages,
  chatStyle: state.ui.chatStyle,
});
const messageListMapDispatchToProps = dispatch => ({ });

export default connect(
  messageListMapStateToProps,
  messageListMapDispatchToProps,
)(MessageList);
