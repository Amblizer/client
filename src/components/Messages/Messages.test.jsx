import React from 'react';

// import { describe, it, expect, jest } from 'jest';
import { shallow } from 'enzyme';
// import { snapshot } from 'react-test-renderer';

import Messages from './Messages.jsx';

const socket = {
  on: jest.fn(),
  emit: jest.fn(),
};

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    ui: { chatStyle: 'TEST' },
    chat: { messages: [] },
  })),
};

describe('MessageList', () => {
  it('matches snapshot', () => {
    const component = shallow(<Messages store={store} socket={socket} />);
    expect(component).toMatchSnapshot();
  });
});
