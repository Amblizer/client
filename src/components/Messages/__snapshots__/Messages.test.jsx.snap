exports[`MessageList matches snapshot 1`] = `
<Component
  chatStyle="TEST"
  messages={Array []}
  socket={
    Object {
      "emit": [Function],
      "on": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
